### DISCUSSION TITLE
WindowsAppRuntime installation error: The package could not be installed because a higher version of this package is already installed.

### DISCUSSION THREAD

**Discussion by rcohn** [microsoft/WindowsAppSDK#2708] [Q&A] [answered]
Hello -

Is there a way around the problem in which WindowsAppRuntime will not install because a higher version of that runtime is already installed?  

Here is the error I see:
`c:\temp\Microsoft.WindowsAppRuntime.Redist1.1.0>windowsappruntimeinstall --license
INFO: Provisioning of WindowsAppSDK packages will be skipped as it requires elevation.
Installing license: MSIX_MAINPACKAGE_LICENSE
Install result : 0x0
Installing license: MSIX_SINGLETONPACKAGE_LICENSE
Install result : 0x0
Deploying package: Microsoft.WindowsAppRuntime.1.1_1000.516.2156.0_x64__8wekyb3d8bbwe
Package deployment result : 0x80073d06 The package could not be installed because a higher version of this package is already installed.
One or more install operations failed. Result: 0x80073d06`

This is problematic because a previously installed app with a higher WindowsAppRuntime version would prevent a lower version from installing.  And the WindowsAppSDK for that lower version is what was built with. 

What would you suggest as a solution to always make sure the version needed by the App is installed?

Thank you.

**Comment by DrusTheAxe** [comment, marked as answer]
>Microsoft.WindowsAppRuntime.1.1_1000.516.2156.0_x64__8wekyb3d8bbwe Package deployment result : 0x80073d06 The package could not be installed because a higher version of this package is already installed. One or more install operations failed. Result: 0x80073d06

That's a bug - it should be OK, not an error.

If v1.1.3 is installed and you run the installer for v1.1.2 we call `packageManager.AddPackageAsync(...v1.1.2...)` which reutrns 0x80073d06 == `ERROR_INSTALL_PACKAGE_DOWNGRADE` which is technically an error but functionally _it's OK_. If you build with 1.1.2 we don't require the runtime packages exactly match 1.1.2, but rather that there needs to be a package equal-or-higher than 1.1.2 (i.e. `>=1.1.2` not `==1.1.2`). If 1.1.3 is installed then all is well.

Problem is we should be trapping `ERROR_INSTALL_PACKAGE_DOWNGRADE` **and treating it as success**.

That's a bug on our part. The fix seems easy but checking with @sachintaMSFT to make sure the right bits are tickled. Will have a PR up later today (maybe even a complete one :P) and will aim to have it in main and the next servicing update.

Thanks for bringing this to our attention!

**Reply by rcohn** [reply]
Thanks for the reply @DrusTheAxe -

Actually, we decide to go around this problem by using `<WindowsAppSDKSelfContained>`.  So far (fingers crossed), this seems to be working out OK and insulating our installation from installed versions of WindowsAppRuntime.

But it's good to know that if the self-contained approach does not work out for some reason, that future versions of WindowsAppRuntime will be fixed against the bug I reported.

Thanks again.

**Reply by DrusTheAxe** [reply]
PR [WindowsAppRuntimeInstaller.exe fails if newer version present #2723](https://github.com/microsoft/WindowsAppSDK/pull/2723)

**Comment by gabbybilka** [comment]
This issue was fixed and released in Windows App SDK 1.1.3 - check out the [release notes](https://aka.ms/windowsappsdk/1.1/1.1.3/release-notes) and [NuGet](https://aka.ms/windowsappsdk/1.1/1.1.3) to try it out 🎉
